src/Skybrud.Umbraco.Redirects/Services/RedirectsService.cs

```diff
@@ -453,19 +459,46 @@ public class RedirectsService : IRedirectsService {
         /// Returns an array of all rode nodes configured in Umbraco.
         /// </summary>
         /// <returns>An array of <see cref="RedirectRootNode"/> representing the root nodes.</returns>
-        public RedirectRootNode[] GetRootNodes()  {
-
-            // Multiple domains may be configured for a single node, so we need to group the domains before proceeding
-            var domainsByRootNodeId = GetDomains().GroupBy(x => x.RootNodeId);
-
-            return (
-                from domainGroup in domainsByRootNodeId
-                let content =  _contentService.GetById(domainGroup.First().RootNodeId)
-                where content is { Trashed: false }
-                orderby content.Id
-                select RedirectRootNode.GetFromContent(content, domainGroup)
-            ).ToArray();
+        public RedirectRootNode[] GetRootNodes()
+        {
+	        var domainsByRootNodeId = GetDomains().GroupBy(x => x.RootNodeId);
+
+	        return (
+		        from domainGroup in domainsByRootNodeId
+		        let content = _contentService.GetById(domainGroup.First().RootNodeId)
+		        where content is { Trashed: false }
+		        orderby content.Id
+		        select RedirectRootNode.GetFromContent(content, domainGroup)
+	        ).ToArray();
+        }
 
+        /// <summary>
+        /// Returns an array of all rode nodes configured in Umbraco.
+        /// </summary>
+        /// <param name="user">An <see cref="IUser"/> with potential root node access restrictions.</param>
+        /// <returns>An array of <see cref="RedirectRootNode"/> representing the root nodes the user has access to.</returns>
+        public RedirectRootNode[] GetRootNodes(IUser user)
+        {
+	        var rootNodes = GetRootNodes();
+	        HashSet<int> rootNodeIds = new();
+
+	        if (user.StartContentIds != null)
+	        {
+		        foreach (var rootNodeId in user.StartContentIds)
+		        {
+			        rootNodeIds.Add(rootNodeId);
+		        }
+	        }
+
+	        foreach (var group in user.Groups)
+	        {
+		        if (group.StartContentId != null)
+		        {
+			        rootNodeIds.Add(group.StartContentId.Value);
+		        }
+	        }
+
+	        return rootNodes.Where(rootNode => rootNodeIds.Any(x => rootNode.Path.Contains(x))).ToArray();
```

Comment:
This is still a breaking change as you're introducing a new method. It's probably not very likely, but developers may choose to write their own implementation of the `IRedirectsService` interface, and if they do, they'll now be missing the `GetRootNodes` method.

C# 8 introduces [default interface methods](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods). So as the method doesn't use any dependencies, you should be able to move the method body to the interface instead. This ensures that we can add a new method without it being a breaking change.

Hope that makes sense 😉